### DISCUSSION TITLE
New release in 2024

### DISCUSSION THREAD

**Discussion by denisvmedia** [golangci/golangci-lint#4312] [Q&A] [answered]
Hi,

I'm wondering, should we have a new release before go 1.22 is released? Since, when it's released, we'll have to wait more for all the support for 1.22 updates (dependent linters will probably also delay the update). In particular, there was a number of updates and bug fixes in `revive`, which would be nice to see in `golangci-lint`.

Thanks!

**Comment by ldez** [comment, marked as answer]
> should we have a new release before go 1.22 is released?

I think we will wait the the release of go1.22.

**Reply by skrutiy** [reply]
Is there any flexibility on this? I've been hoping to use the `goconst` `ignore-strings` feature, which has now been on the docs for `golangci` for a couple months (#4200): https://golangci-lint.run/usage/linters/#goconst, but not actually supported in the latest release. I understand if the priority still is to wait for go 1.22, but wanted to bring to attention that the upstream docs show a feature that can not actually be used yet.

**Reply by ldez** [reply]
golangci-lint project is handled with one branch, and when something is fixed in the doc or added in the doc it's available immediately, so yes the doc is always "ready" before the release.
This is the best compromise to being able to fix the doc without maintaining multiple branches.
So The fact that the doc already displays new features is not something that influences whether we should create a release or not.

**Comment by ldez** [comment]
The release [v1.56.0](https://github.com/golangci/golangci-lint/releases/tag/v1.56.0) is available.
